### DISCUSSION TITLE
mqtt keepalive

### DISCUSSION THREAD

**Discussion by GMagician** [blakeblackshear/frigate#14048] [Ask A Question]
Am I wrong or mqtt keepalive is fixed at 60s?

**Comment by Unspec7** [comment]
It's hardcoded into Frigate.

https://github.com/blakeblackshear/frigate/blob/9c751230a1be643ce7c43897c535d5df29947836/frigate/comms/mqtt.py#L254

From paho's documentation:

connect_async(host: str, port: int = 1883, keepalive: int = 60, bind_address: str = '', bind_port: int = 0, clean_start: bool | Literal[3] = 3, properties: [Properties](https://eclipse.dev/paho/files/paho.mqtt.python/html/types.html#paho.mqtt.properties.Properties) | None = None)

Seems extremely simple to make the timeout configurable. Gonna see if I can PR a change to make it configurable via a keepalive config option.
